pkg/api/mlflow/dao/repositories/param.go

```diff
@@ -31,8 +32,50 @@ func NewParamRepository(db *gorm.DB) *ParamRepository {
 
 // CreateBatch creates []models.Param entities in batch.
 func (r ParamRepository) CreateBatch(ctx context.Context, batchSize int, params []models.Param) error {
-	if err := r.db.CreateInBatches(params, batchSize).Error; err != nil {
+	var err error
+	err = r.db.CreateInBatches(params, batchSize).Error
+	if err != nil {
+		// remove duplicate rows and try again
+		params, err1 := r.removeExactMatches(ctx, params)
+		if err1 != nil {
+			return eris.Wrap(err1, "error removing exact matches after: " + err.Error())
+		}
+		err = r.db.CreateInBatches(params, batchSize).Error
+	}
+
+	if err != nil {
 		return eris.Wrap(err, "error creating params in batch")
 	}
 	return nil
 }
+
+// removeExactMatches will return a new slice of params which excludes exact matches
+func (r ParamRepository) removeExactMatches(ctx context.Context, params []models.Param) ([]models.Param, error) {
+	var keys, foundKeys []string
+	paramMap := map[string]models.Param{}
+	paramsToReturn := []models.Param{}
+	for _, param := range params {
+		key := fmt.Sprintf("%v-%v-%v", param.RunID, param.Key, param.Value)
+		keys = append(keys, key)
+		paramMap[key] = param
+	}
+
+	tx := r.db.Raw(`
+           select run_uuid || '-' || key || '-' || value
```

Comment:
@jgiannuzzi @suprjinx  what about next solution -> when we create a param, lets calculate a `hash` into the own column and then let's try to use this hash to find duplicated.

Reply:
interesting idea

Reply:
I can see one problem here - how it should work for old records without the hash? maybe we can create a command line tool, which will migrate old data and add missing hashes?

Reply:
I do like this idea, especially because we could then use `ON CONFLICT(hash_column) DO NOTHING` and make things much simpler. Perhaps old records wouldn't be getting new log params, so maybe it's okay without backfill?

Reply:
decided to keep this in the error-only path